### PR TITLE
add coinchange implementation

### PR DESCRIPTION
**Pull request by nursyah21**

### CODE REVIEW

File: dynamic/coinchange.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package dynamic
+/*
```
Comment (raklaptudirm) [inline dynamic/coinchange.go:2]: Looks like you documented the code using jsdoc. Go uses a much simpler way to document code. https://go.dev/doc/effective_go#commentary

File: dynamic/coinchange.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@params 
+coins: combination of coin 
+amount: target coins
+
+@return
+num of combination
+ */
+
+
+func coinChange(coins []int32, amount int32) int32 {
```
Comment (raklaptudirm) [inline dynamic/coinchange.go:12]: Export this function:
```suggestion
func CoinChange(coins []int32, amount int32) int32 {
```

File: dynamic/coinchange_test.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package dynamic
```
Comment (raklaptudirm) [inline dynamic/coinchange_test.go:1]: Use black-box testing:
```suggestion
package dynamic_test
```

**Review by raklaptudirm — CHANGES_REQUESTED** [review]
Format your code with `gofmt`

File: dynamic/coinchange_test.go (hunk 3 above)
Comment (nursyah21) [reply dynamic/coinchange_test.go:1]: if i use dynamic_test, the output is
./coinchange_test.go:23:14: undefined: CoinChange
FAIL	github.com/TheAlgorithms/Go/dynamic [build failed]

how do i run code, if i use black-box testing?

Comment (raklaptudirm) [reply dynamic/coinchange_test.go:1]: You first need to change the function name to export it.

**Pull request closed without merging**